### PR TITLE
[work in progress] Dropping Python 2 support

### LINKED ISSUE #285: Drop Python 2 support
Python 2 is [retiring soon](https://pythonclock.org/), and many projects are [dropping support for Python 2.7](https://python3statement.org/).

Is is time to drop support of Python 2 and upgrade the package requirements?

### PR DESCRIPTION
**Pull request by tommyod**
This PR closes #285 .

- Removed Python 2 from Travis CI testing.
- Removed the entire `externals` folder and associated code.
- Removed `base.py`, importing from sklearn instead.

### CODE REVIEW

File: pyglmnet/base.py [hunk 1]
```diff
@@ -1,171 +0,0 @@
-import warnings
```
Comment (jasmainak) [inline pyglmnet/base.py]: please don't remove this. We don't want a dependency on sklearn. It's a conscious choice.

File: pyglmnet/pyglmnet.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 from scipy.special import expit
 from scipy.stats import norm
 from .utils import logger, set_log_level
-from .base import BaseEstimator, is_classifier, check_version
+from sklearn.base import BaseEstimator, is_classifier
```
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]: see above. We don't want a dependency on sklearn.

File: pyglmnet/pyglmnet.py [hunk 3]
```diff
@@ -881,7 +863,7 @@ def score(self, X, y):
 
 
 class GLMCV(object):
-    """Class for estimating regularized generalized linear models (GLM)
+    r"""Class for estimating regularized generalized linear models (GLM)
```
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]: why do you need this? It's not related to python 2, is it?

Comment (tommyod) [reply pyglmnet/pyglmnet.py]: Running tests gave warnings [like this](https://github.com/pallets/jinja/issues/646). It's unrelated to Python 2.

**Comment by tommyod** [comment]
Spent some time reverting the commits related to sklearn - I'm not proud of the resulting git history. Please squash when merging.

I will update docs.

File: pyglmnet/externals/__init__.py [hunk 4]
```diff
@@ -1,2 +1 @@
-from . import six
```
Comment (jasmainak) [inline pyglmnet/externals/__init__.py]: you can also remove six, no?

**Comment by jasmainak** [comment]
Can you also update the Readme and documentation saying we only support python 3 from now on.

**Comment by tommyod** [comment]
I think the PR is good to go, can you look over it @jasmainak ? There were some issues with CircleCI, and on the webpage is states "_Your project references CircleCI 1.0 or it has no configuration. CircleCI 1.0 and projects without configuration files are no longer supported. You must update your project to use CircleCI 2.0 configuration to continue._" I'm not experienced with CircleCI, but I suggest we fix it in another PR if troubles persist.

Please squash merge, the history is very ugly.

**Comment by jasmainak** [comment]
Done, thanks @tommyod 

next time I suggest when you are ready you can put [MRG] at the beginning of the title. It's how we usually indicate that it's good to go :)

**Pull request merged**